tests/test_trajectory.py

```diff
@@ -0,0 +1,305 @@
+                                                           well_name = well_name,
+                                                           length_uom = 'm',
+                                                           md_datum = datum)
+            # -------- Assert ---------
+            assert trajectory_from_ascii is not None
+
+
+def test_set_tangents(example_model_and_crs):
+    # --------- Arrange ----------
+    model, crs = example_model_and_crs
+    elevation = 100
+    # Create a measured depth datum
+    datum = resqpy.well.MdDatum(parent_model = model,
+                                crs_uuid = crs.uuid,
+                                location = (0, 0, -elevation),
+                                md_reference = 'kelly bushing')
+    mds = np.array([300, 310, 330, 340])
+    zs = mds - elevation
+    well_name = 'Coconut'
+    source_dataframe = pd.DataFrame({
+        'MD': mds,
+        'X': [1, 2, 3, 4],
+        'Y': [1, 2, 3, 4],
+        'Z': zs,
+        'WELL': ['Coconut', 'Coconut', 'Coconut', 'Coconut']
+    })
+
+    # Create a trajectory from dataframe
+    trajectory = resqpy.well.Trajectory(parent_model = model,
+                                        data_frame = source_dataframe,
+                                        well_name = well_name,
+                                        md_datum = datum,
+                                        length_uom = 'm')
+
+    # --------- Act ----------
+    # Calculate tangent vectors based on control points
+    trajectory.set_tangents()
+
+    # --------- Assert ----------
+    assert trajectory.tangent_vectors is not None
+    assert trajectory.tangent_vectors.shape == (trajectory.knot_count, 3)
+
+
+def test_xyz_for_md(example_model_and_crs):
+
+    # --------- Arrange ----------
+    model, crs = example_model_and_crs
+    elevation = 100
+    # Create a measured depth datum
+    datum = resqpy.well.MdDatum(parent_model = model,
+                                crs_uuid = crs.uuid,
+                                location = (0, 0, -elevation),
+                                md_reference = 'kelly bushing')
+    mds = np.array([300, 310, 330, 340])
+    zs = mds - elevation
+    well_name = 'Coconut'
+    source_dataframe = pd.DataFrame({
+        'MD': mds,
+        'X': [1, 2, 3, 4],
+        'Y': [1, 2, 3, 4],
+        'Z': zs,
+        'WELL': ['Coconut', 'Coconut', 'Coconut', 'Coconut']
+    })
+
+    # Create a trajectory from dataframe
+    trajectory = resqpy.well.Trajectory(parent_model = model,
+                                        data_frame = source_dataframe,
+                                        well_name = well_name,
+                                        md_datum = datum,
+                                        length_uom = 'm')
+    # --------- Act ----------
+    # # Get the xyz triplet for a given measured depth
+    x, y, z = trajectory.xyz_for_md(md = 305)
+
+    # # -------- Assert ---------
+    assert x == y == 1.5
```

Comment:
better to use maths.isclose() rather than an exact == when testing computed floating point values
(where maths is what I import math as)